components/automate-cli/pkg/verifysystemdcreate/systemdcreatesvc.go

```diff
@@ -0,0 +1,157 @@
+package verifysystemdcreate
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/chef/automate/components/automate-deployment/pkg/cli"
+	"github.com/pkg/errors"
+)
+
+const (
+	SERVICE_NAME        = "automate-verify"
+	SERVICE_DESCRIPTION = "Service for automating verification"
+	SERVICE_COMMAND     = "chef-automate verify serve"
+	SYSTEMD_FILE        = "%s.service"
+	TEMPLATE_NAME       = "systemd-service"
+	TEMPLATE_TEXT       = `[Unit]
+Description={{.ServiceDescription}}
+After=network.target
+
+[Service]
+ExecStart={{.ServiceCommand}}
+Restart=always
+StandardOutput=journal
+StandardError=journal
+
+[Install]
+WantedBy=multi-user.target
+`
+)
+
+type systemdInput struct {
+	ServiceName        string
+	ServiceDescription string
+	ServiceCommand     string
+}
+
+type CreateSystemdService struct {
+	SystemdCreateUtils      SystemdCreateUtils
+	BinaryDestinationFolder string
+	SystemdLocation         string
+	Writer                  *cli.Writer
+}
+
+func NewCreateSystemdService(
+	systemdCreateUtils SystemdCreateUtils,
+	binaryDestinationFolder string,
+	systemdLocation string,
+	writer *cli.Writer) (*CreateSystemdService, error) {
+	if binaryDestinationFolder == "" {
+		return nil, errors.New("Binary destination folder cannot be empty")
+	}
+	if systemdLocation == "" {
+		return nil, errors.New("Systemd location cannot be empty")
+	}
```

Comment:
Any reason for not validating `systemdCreateUtils` and  `writer`?

Reply:
added nil check